zson/marshal.go

```diff
@@ -290,6 +290,11 @@ func (m *MarshalZNGContext) encodeValue(v reflect.Value) (zng.Type, error) {
 		kind := v.Kind().String()
 		if name != "" && name != kind {
 			path := v.Type().PkgPath()
+			// We do not want to further decorate nano.Ts as
+			// it's already been converted to a Zed time.
+			if strings.HasSuffix(path, "/nano") && name == "Ts" {
```

Comment:
This doesn't feel quite right. Could we say `if v.Type() == nanoTsType {` here (with `var nanoTsType = reflect.TypeOf(nano.Ts(0))` at the top level)?

Reply:
Great idea.